service/src/main/java/com/sedliarov/learningtable/service/impl/StudentServiceImpl.java

```diff
@@ -28,31 +31,81 @@ public class StudentServiceImpl implements StudentService {
   @Override
   public StudentDto createStudent(StudentDto studentDto) {
     Student studentToSave = mapper.dtoToEntity(studentDto);
+    Student studentToCheck = repository.findBySecondName(studentToSave.getSecondName());
+    if (studentToCheck != null) {
+      throw new EntityAlreadyExistsException("Entity already exist");
+    }
     return mapper.entityToDto(repository.save(studentToSave));
   }
 
   @Override
   public StudentDto updateStudent(UUID id, StudentDto studentDto) {
-    repository.findById(id).orElseThrow(() -> new EntityNotFoundException("Entity not found with id " + id));
-
+    repository.findById(id).orElseThrow(()
+        -> new UpdateEntityNotFoundException("Entity for update not found"));
     Student studentToUpdate = mapper.dtoToEntity(studentDto);
     studentToUpdate.setStudentId(id);
     return mapper.entityToDto(repository.save(studentToUpdate));
   }
 
   @Override
   public void deleteStudent(UUID id) {
+    repository.findById(id).orElseThrow(()
+        -> new DeleteEntityNotFoundException("Entity not delete because not existed"));
     repository.deleteById(id);
   }
 
   @Override
   public StudentDto getStudentById(UUID id) {
-    return mapper.entityToDto(repository.findById(id)
-        .orElseThrow(() -> new EntityNotFoundException("Entity not found with id " + id)));
+    return mapper.entityToDto(repository.findById(id).orElseThrow(()
+        -> new CustomEntityNotFoundException("Entity not found with id " + id)));
   }
 
   @Override
   public List<StudentDto> getStudents() {
     return repository.findAll().stream().map(mapper::entityToDto).toList();
   }
+
+  /**
+   * Exception for GET response if student not found.
+   */
+  @ResponseStatus(value = HttpStatus.NOT_FOUND)
+  public class CustomEntityNotFoundException extends EntityNotFoundException {
+
+    public CustomEntityNotFoundException(String message) {
+      super(message);
+    }
+  }
+
+  /**
+   * Exception for DELETE response if student not found.
+   */
+  @ResponseStatus(value = HttpStatus.BAD_REQUEST)
+  public class DeleteEntityNotFoundException extends EntityNotFoundException {
+
+    public DeleteEntityNotFoundException(String message) {
+      super(message);
+    }
+  }
+
+  /**
+   * Exception for UPDATE response if student not found.
+   */
+  @ResponseStatus(value = HttpStatus.BAD_REQUEST)
+  public class UpdateEntityNotFoundException extends EntityNotFoundException {
+
+    public UpdateEntityNotFoundException(String message) {
+      super(message);
+    }
+  }
+
+  /**
+   * Exception for POST response if student already exist.
+   */
+  @ResponseStatus(value = HttpStatus.BAD_REQUEST)
+  public class EntityAlreadyExistsException extends HttpMessageNotWritableException {
+
+    public EntityAlreadyExistsException(String message) {
+      super(message);
+    }
+  }
```

Comment:
Hold on, hold on! I guess is not super class of exception

Reply:
see https://www.baeldung.com/exception-handling-for-rest-with-spring

Reply:
Done